go/obsclient/authclient.go

```diff
@@ -77,7 +77,7 @@ func (ac *AuthObsClient) NonceAt(ctx context.Context, blockNumber *big.Int) (uin
 
 func (ac *AuthObsClient) CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error) {
 	var hex string
-	err := ac.rpcClient.CallContext(ctx, &hex, rpc.RPCCall, toCallArg(msg), toBlockNumArg(blockNumber))
+	err := ac.rpcClient.CallContext(ctx, &hex, rpc.RPCCall, msg, toBlockNumArg(blockNumber))
```

Comment:
`toCallArg` loses data, also the underlying msg data will be encrypted (and thus 
marshalled)

Reply:
This worries me, because the geth client uses toCallArg before it marshals (I should have left a comment to say it's copy-pasted from geth). They must have had a reason.

Reply:
Good point - I reviewed it and instead of using a json serializer in the ethereum.callMsg. They manually serialize the fields.
Reverted 👍